### DISCUSSION TITLE
“Regular” Restful routes

### DISCUSSION THREAD

**Discussion by dangdennis** [encoredev/encore#34] [Help] [answered]
I understand encore is geared towards api-only microservices style, but is it possible to still do html templating with restful routes?

I’d like to serve a frontend from the same system.

**Comment by eandre** [comment, marked as answer]
**Update September 2021**: Encore now ships with first-class REST API support 🎉! [Check out the docs](https://encore.dev/docs/develop/services-and-apis#rest-apis) to get started :)

Hey @dangdennis, you can certainly do HTML templating from your API routes by dropping down to a "raw handler":

```go
package myservice

//encore:api public raw
func MyHandler(w http.ResponseWriter, req *http.Request) {
    tmpl.Execute(w, nil) // or whatever
}
```

And then you can call this endpoint in the same way as other endpoints, with path `/myservice.MyHandler`

The other aspect of your question is around RESTFul routes, which we don't have good support for today. It's something we definitely want to add, we just need to figure out the best way to design it. I have some ideas but still thinking through the different options.

**Reply by dangdennis** [reply]
I figured that's the case. Thanks @eandre.

**Reply by eandre** [reply]
Hey @dangdennis, just to follow up, Encore now ships with built-in support for RESTful APIs ([see docs](https://encore.dev/docs/develop/services-and-apis#rest-apis)). Thanks for your feedback!

**Reply by dangdennis** [reply]
Woah ho ho! And you got back to me. Thanks @eandre
